analyzer/src/main/kotlin/managers/utils/NodeSupport.kt

```diff
@@ -125,35 +142,50 @@ fun expandNpmShortcutUrl(url: String): String {
 }
 
 private val NPM_LOCK_FILES = listOf("npm-shrinkwrap.json", "package-lock.json")
+private val PNPM_LOCK_FILES = listOf("pnpm-lock.yaml")
 private val YARN_LOCK_FILES = listOf("yarn.lock")
 
 private data class PackageJsonInfo(
     val definitionFile: File,
     val hasYarnLockfile: Boolean = false,
     val hasYarn2ResourceFile: Boolean = false,
     val hasNpmLockfile: Boolean = false,
+    val hasPnpmLockfile: Boolean = false,
+    val isPnpmWorkspaceRoot: Boolean = false,
+    val isPnpmWorkspaceSubmodule: Boolean = false,
     val isYarnWorkspaceRoot: Boolean = false,
     val isYarnWorkspaceSubmodule: Boolean = false
 )
 
+private fun isHandledByPnpm(entry: PackageJsonInfo) =
```

Comment:
previously, isHandledByYarn and isHandledByNpm have been mutually exclusive.
Now, with the three functions, they are not mutually excusive anymore. 
Have you considered that, e.g. isHandledByPnpm and isHandledByYarn return true both? Can it lead to problems?

Reply:
This is decided based on the content of this object:
```kt
PackageJsonInfo(
    definitionFile = definitionFile,
    isPnpmWorkspaceRoot = isPnpmWorkspaceRoot(definitionFile.parentFile),
    isYarnWorkspaceRoot = isYarnWorkspaceRoot(definitionFile) && !isPnpmWorkspaceRoot(definitionFile.parentFile),
    hasYarnLockfile = hasYarnLockFile(definitionFile.parentFile),
    hasNpmLockfile = hasNpmLockFile(definitionFile.parentFile),
    hasPnpmLockfile = hasPnpmLockFile(definitionFile.parentFile),
    hasYarn2ResourceFile = hasYarn2ResourceFile(definitionFile.parentFile),
    isPnpmWorkspaceSubmodule = definitionFile in pnpmWorkspaceSubmodules,
    isYarnWorkspaceSubmodule = definitionFile in yarnWorkspaceSubmodules && definitionFile !in pnpmWorkspaceSubmodules
)
```

There are three cases when it is unclear if the project is managed by Yarn or by PNPM:
1. If the PNPM project also stores the workspaces in the root package.json, then `yarnWorkspaceSubmodules` will contain actual PNPM submodules. I worked around that by checking both:
`isYarnWorkspaceSubmodule = definitionFile in yarnWorkspaceSubmodules && definitionFile !in pnpmWorkspaceSubmodules`
2. Same as (1) for the `workspaceRoot`:
`isYarnWorkspaceRoot = isYarnWorkspaceRoot(definitionFile) && !isPnpmWorkspaceRoot(definitionFile.parentFile),`
3. If the directory contains both, a `yarn.lock` file and a `pnpm-lock.yaml` file. In this case, ORT would try to analyze the project with both package managers. Do you have a suggestion how to handle that case?

Reply:
The handling make sense to me, thanks! Regarding point #3 I don't have any proposal either.